### PR TITLE
Union docs

### PR DESCRIPTION
**Pull request by samuelcolvin**
Move union docs into their own section in "Concepts", rewrite the beginning of union docs.

### CODE REVIEW

File: docs/api/standard_library_types.md [hunk 1]
```diff
@@ -774,356 +774,7 @@ In case you want to constrain the UUID version, you can check the following type
...
-    Model(pet={'pet_type': 'cat', 'color': 'black'}, n='1')
-except ValidationError as e:
-    print(e)
-    """
-    1 validation error for Model
-    pet.cat.black.black_name
-      Field required [type=missing, input_value={'pet_type': 'cat', 'color': 'black'}, input_type=dict]
-    """
-```
+Pydantic has extensive support for union validation, both `typing.Union` and Python 3.10's pipe syntax (`A | B`) are supported.
```
Comment (sydney-runkle) [inline docs/api/standard_library_types.md:777]: Could we add a link to the concepts/unions page here?

**Review by sydney-runkle — APPROVED** [review]
LGTM otherwise 👍
I think this change will help a lot - I think people are less likely to report easy to answer issues if we have this front and center on the `concepts` page

**Comment by sydney-runkle** [comment]
I'll pull this down and add the link to the other union docs + fix the tests.

File: docs/api/standard_library_types.md (hunk 1 above)
Comment (samuelcolvin) [reply docs/api/standard_library_types.md:777]: oh yes, sorry that should be first.

**Pull request merged**